File: trees/traversal/TreeTraversal.java
package trees.traversal;

import trees.TreeNode;

import java.util.ArrayList;

public interface TreeTraversal {
    ArrayList<Integer> preOrderTraversal(TreeNode root);
    ArrayList<Integer> inOrderTraversal(TreeNode root);
    ArrayList<Integer> postOrderTraversal(TreeNode root);
}

File: src/main/kotlin/dev/mslalith/trees/traversal/TreeTraversal.kt
package dev.mslalith.trees.traversal

import dev.mslalith.trees.TreeNode

interface TreeTraversal {
    fun preOrderTraversal(root: TreeNode?): List<Int>
    fun inOrderTraversal(root: TreeNode?): List<Int>
    fun postOrderTraversal(root: TreeNode?): List<Int>
}
